src/main/java/buildcraftAdditions/proxy/ClientProxy.java

```diff
@@ -59,4 +62,8 @@ public EntityPlayer getPlayer(String name) {
 	public void cloneFluidTextures(Fluid source, Fluid target) {
 		target.setIcons(source.getStillIcon(), source.getFlowingIcon());
 	}
+
+	public static GuiScreen getCurrentOpenedScreen() {
+		return Minecraft.getMinecraft().currentScreen;
```

Comment:
as i said earlier, you don't need this static, better is to call this from the proxy instance and for that the static isn't needed, also what heapened to the isShiftDown()?

and this method should be in the commonProxy, there it return null and here it should override the method and return the correct thing